modules/@apostrophecms/i18n/index.js

```diff
@@ -227,6 +324,15 @@ module.exports = {
           self.getComponentName('localizeModal', 'AposI18nLocalize'),
           { moduleName: self.__meta.name }
         );
+      },
+      setPrefixUrls(req) {
+        // For bc, req.baseUrl is always set, to a best guess if baseUrl is not configured.
+        // When falling back, req.hostname is used if trustProxy is active, otherwise the
+        // Host header to allow port numbers in dev
+        req.baseUrl = self.apos.page.getBaseUrl(req) || (req.protocol + '://' + (self.options.trustProxy && req.get('X-Forwarded-Host')) ? req.hostname : req.get('Host'));
```

Comment:
I'd prefer you get the hostname as a step before to make this easier to parse. Then it could be cleanly written with a template literal.